internal/server/middlewares.go

```diff
@@ -124,6 +124,10 @@ func requestLogger(lg *zap.Logger) middleware {
 				}{wrapped, flusher}
 			}
 
+			buf, _ := io.ReadAll(req.Body)
```

Comment:
@ishanarya0  this is hard to understand without context.
we can do it like below sample

```go
bodyCopy := bytes.NewBuffer(nil)
noOfBytes, err := io.Copy(bodyCopy, req.Body)
if err != nil {
	lg.Error("error copying request body: %v", zap.String("error", err.Error()))
}

next.ServeHTTP(fr, req)

if (noOfBytes > 0) {
	...log
}
```